strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/AuthenticatorsConfigController.java

```diff
@@ -28,63 +25,72 @@
     }
 
     @ApiOperation(value = "Reorders authenticators by their indexes")
-    @ApiResponses(value = { @ApiResponse(code = 200,
-            message = "Reorder succeeded"),
-                            @ApiResponse(code = 500,
-                                    message = "Reorder failed") })
-    @RequestMapping(value = "/reorder/{first}/{second}",
-            method = RequestMethod.PUT)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = SUCCESSFUL_REORDER),
+                            @ApiResponse(code = 400, message = FAILED_REORDER) })
+    @PutMapping(value = "/reorder/{first}/{second}",
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = { MediaType.TEXT_PLAIN_VALUE,
+                         MediaType.APPLICATION_JSON_VALUE })
     public ResponseEntity reorder(@PathVariable int first,
-                                  @PathVariable int second)
+                                  @PathVariable int second,
+                                  @RequestHeader(HttpHeaders.ACCEPT) String acceptHeader)
     {
         try
         {
             authenticatorsRegistry.reorder(first, second);
-            return ResponseEntity.ok("Reorder succeeded");
+            return getSuccessfulResponseEntity(SUCCESSFUL_REORDER, acceptHeader);
         }
         catch (Exception e)
         {
-            logger.error("Error during reorder processing.", e);
-            return toError("Error during reorder processing: " + e.getLocalizedMessage());
+            return getBadRequestResponseEntity(FAILED_REORDER, acceptHeader);
         }
     }
 
     @ApiOperation(value = "Reloads authenticators registry")
-    @ApiResponses(value = { @ApiResponse(code = 200,
-            message = "Reload succeeded"),
-                            @ApiResponse(code = 500,
-                                    message = "Reload failed") })
-    @RequestMapping(value = "/reload",
-            method = RequestMethod.PUT)
-    public ResponseEntity reload()
+    @ApiResponses(value = { @ApiResponse(code = 200, message = SUCCESSFUL_RELOAD),
+                            @ApiResponse(code = 400, message = FAILED_RELOAD) })
+    @PutMapping(value = "/reload",
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = { MediaType.TEXT_PLAIN_VALUE,
+                         MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity reload(@RequestHeader(HttpHeaders.ACCEPT) String acceptHeader)
     {
         try
         {
             authenticatorsScanner.scanAndReloadRegistry();
-
-            return ResponseEntity.ok("Reload succeeded");
+            return getSuccessfulResponseEntity(SUCCESSFUL_RELOAD, acceptHeader);
         }
         catch (Exception e)
         {
-            logger.error("Error during reload processing.", e);
-            return toError("Error during reload processing: " + e.getLocalizedMessage());
+            return getBadRequestResponseEntity(FAILED_RELOAD, acceptHeader);
```

Comment:
Same as the above comment regarding `getExceptionResponseEntity`. In addition, I believe this is one of those exceptional cases where you should actually return `Internal Server Error`, because the client won't actually be sending any data to the server and the only possible error that can occur is related to something internally handled (i.e. xml file containing the authenticators is broken or something else which is beyond the client's reach).